### PR TITLE
Move remaining client functions in components to actions

### PR DESCRIPTION
**Pull request by cpanato**
WORK IN PROGRESS 
#### Summary
Move remaining  components to an action

#### Ticket Link
N/A

@jwilander please review :) a lot of stuffs

### CODE REVIEW

**Comment by jwilander** [comment]
I'll take a look a little later, thanks @cpanato !

File: webapp/actions/user_actions.jsx [hunk 1]
```diff
@@ -616,3 +636,142 @@ export function resendVerification(email, success, error) {
...
     );
 }
+
+export function loginById(userId, password, mfaToken, success, error) {
+    Client.loginById(
+        userId,
+        password,
+        mfaToken,
+        () => {
+            if (this.state.hash > 0) {
```
Comment (jwilander) [inline webapp/actions/user_actions.jsx]: I think you need to pass `hash` in here since you can't access `this.state`

**Comment by jasonblais** [comment]
Tested the following with a System Admin account on Windows 10, Chrome:

- Revoking a session
- Updating the config from the system console
- Reloading config from disk in the system console
- Purging caches in the system console
- Test email button in the system console
- Uploading a SAML cert in the system console
- Adding an OAuth app
- Switch account types
- Upload/remove license file
- Logging in with MFA
- Uploading profile picture


Found a couple of issues:

1) When I go Account Settings > Security > Password, I get the following errors if **System Console > Custom Integrations > Enable OAuth2 Service Provider** is set to false.

image

2) When I revoke a session via Account Settings > Security > View and Logout of Active Sessions, I get the following JavaScript errors

image

**Comment by cpanato** [comment]
@jasonblais I was not able to login in the test server using the test account

**Comment by jasonblais** [comment]
I changed a bunch of things -- let me generate a new server so you can start fresh

**Comment by cpanato** [comment]
@jasonblais for the issue number 2 it happen also in master and this is due this call after we call the revokesessions:
`AsyncClient.getSessions();`
line 11 on the actions/admin_actions.jsx

this is the original behavior, but I not sure if we need make this call since you got disconnected, maybe is just to make sure we are out. Please advise if we can remove that call.


for the issue 1 I will check

**Comment by jasonblais** [comment]
Thanks @cpanato, re-ran the above tests and verified that 1) is fixed and that all other cases work as expected.

Also, confirmed 2) with a dev who previously worked on cleaning server log errors around invalid or expired sessions and manually revoked sessions will cause the error to happen once (it's unavoidable). It's also not critical since the user would only see it if they had the developer console open.

Queuing for dev review,

**Pull request merged**